src/main/java/internet/store/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,148 @@
+package internet.store.dao.jdbc;
+
+import internet.store.dao.OrderDao;
+import internet.store.exception.DataProcessingException;
+import internet.store.lib.Dao;
+import internet.store.model.Order;
+import internet.store.model.Product;
+import internet.store.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class OrderDaoJdbcImpl implements OrderDao {
+
+    @Override
+    public List<Order> getUserOrders(Long userId) {
+        String query = "SELECT order_id FROM orders WHERE deleted = false AND user_id = " + userId;
+        return getOrders(query);
+    }
+
+    @Override
+    public List<Order> getAll() {
+        String query = "SELECT order_id FROM orders WHERE deleted = false;";
+        return getOrders(query);
+    }
+
+    @Override
+    public Order create(Order order) {
+        String query = "INSERT INTO orders (user_id) values (?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setLong(1, order.getUserId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                order.setOrderId(resultSet.getLong(1));
+            }
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Order " + order + " was not created", ex);
+        }
+        return addProductsToOrder(order);
+    }
+
+    @Override
+    public Order update(Order order) {
+        deleteProductsfromOrder(order);
+        return addProductsToOrder(order);
+    }
+
+    @Override
+    public boolean delete(Long orderId) {
```

Comment:
I think in this method we should also clear orders_products table.
Same in other Dao's delete methods

Reply:
You are right! Thanks a lot for advise!

Reply:
I'm sorry, but my comment was wrong. I did it in my project and Roman told me no to do it because after such deletion order become no informative.

Reply:
Here is Roman'comment:
"that's not corrects, because there will be no point in marking order deleted and removing its relations with products. In that case only remained info will be order and user id and it's not informative"